src/wallet/api/wallet.cpp

```diff
@@ -1429,75 +1428,91 @@ PendingTransaction *WalletImpl::createTransaction(const string &dst_addr, const
     PendingTransactionImpl * transaction = new PendingTransactionImpl(*this);
 
     do {
-        if(!cryptonote::get_account_address_from_str(info, m_wallet->nettype(), dst_addr)) {
-            // TODO: copy-paste 'if treating as an address fails, try as url' from simplewallet.cpp:1982
-            setStatusError(tr("Invalid destination address"));
+        bool payment_id_seen = false;
+        std::vector<uint8_t> extra;
+        vector<cryptonote::tx_destination_entry> dsts;
+        if (amount && (dst_addr.size() != (*amount).size())) {
```

Comment:
You replaced the optional with a non optional (in practice). Assuming this is not called elsewhere, why keep the optional ?

Reply:
Can you explain this again? If no amount is provided, `create_transactions_all` gets called. I don’t see how I could do this without optional.

Reply:
I meant the call to that is:

> amount ? (std::vector<uint64_t> {*amount}) : (optional<std::vector<uint64_t>>())

So I missed the optional ctor here, so it's fine :)